requirements-dev.txt

```diff
@@ -1,3 +1,7 @@
+scipy
+numpy
+sympy
+qiskit>=2.1.0
```

Comment:
Are these dev-only requirements?  I think they're more likely runtime requirements - they should be wherever Aer specifies its runtime requirements if so.

I think requiring Qiskit 2.1 would be a big bump for Aer.